File: SMproject/SteamAccountService.cpp
#include "SteamAccountService.h"

SteamAccountData SteamAccountService::LoadAccount(
    const std::string& vanityName)
{
    SteamAccountData accountData;
    SteamAPI steam;
    SkinPortAPI skinport;
    MarketValuation valuationCalculator;

    std::string steamID =
        steam.ResolveVanityURL(vanityName);

    if (steamID.empty())
    {
        return accountData;
    }

    accountData.profile =
        steam.FetchPlayerSummary(steamID);
    accountData.inventory =
        steam.FetchCS2Inventory(steamID);
    std::vector<SkinPortItem> marketData =
        skinport.FetchMarketData();
    accountData.valuation =
        valuationCalculator.CalculateInventoryValue(
            accountData.inventory,
            marketData);
    accountData.loaded = true;

    return accountData;
}

File: AndroidApp/app/src/main/java/com/team01/steamanalyst/service/SteamAccountService.kt
package com.team01.steamanalyst.service

import com.team01.steamanalyst.data.SteamAccountData
import com.team01.steamanalyst.valuation.MarketValuation

class SteamAccountService {

    fun loadAccount(
        vanityName: String,
        apiKey: String
    ): SteamAccountData {

        val steamService = SteamService()
        val skinportService = SkinportService()

        val steamID64 =
            steamService.resolveVanityURL(
                vanityName = vanityName,
                apiKey = apiKey
            )

        val profile =
            steamService.fetchPlayerSummary(
                steamID64 = steamID64,
                apiKey = apiKey
            )

        val inventory =
            steamService.fetchInventory(
                steamID64 = steamID64
            )

        val marketItems =
            skinportService.fetchMarketData()

        val valuation =
            MarketValuation().calculate(
                inventory = inventory,
                marketItems = marketItems
            )

        return SteamAccountData(
            profile = profile,
            inventory = inventory,
            valuation = valuation,
            loaded = true
        )
    }
}